docs/features/server-side-editor-api.md

```diff
@@ -0,0 +1,347 @@
+---
+category: cloud-services
+order: 30
+meta-title: Cloud Services Server-side Editor API | CKEditor 5 Documentation
+meta-description: Learn how to use server-side API to manage content and collaboration data easily without running the editor.
+modified_at: 2025-06-05
+badges: [ premium ]
+---
+
+# Server-side editor API
+
+Server-side Editor API enables deep and complex integration of your application with all document data, enabling you to manipulate content and manage collaborative data such as suggestions, comments, and revision history, and much more, directly from your server-side code.
+
+The remote script REST API endpoint allows you to execute any JavaScript code that uses the CKEditor&nbsp;5 API, that could be executed by a browser, but without a need to open the editor by a human user. Instead, the script is executed on the Cloud Services server.
+
+## Why use server-side editor API?
+
+While CKEditor&nbsp;5 provides a rich client-side editing experience, there are many scenarios where server-side content processing is essential:
+
+* **Automation**: Run content processing tasks as part of your backend workflows.
+* **Scalability**: Process multiple documents simultaneously without client-side limitations.
+* **Security**: Process sensitive content in a controlled environment without exposing it to client-side manipulation.
+* **Performance**: Handle large-scale content operations without impacting the user's browser.
+* **Consistency**: Ensure uniform content changes across multiple documents.
+* **Integration**: Connect with other server-side systems and databases directly.
+
+## Common use cases
+
+* **Deep integration**: Build custom features that can manage document content and related document data straight from your application UI, without a need to open the editor.
+* **Content migration**: Restructure and update references across multiple documents, perfect for website redesigns or content reorganization.
+* **Shared content blocks**: Automatically update reusable content (like headers, footers, or common sections) across all documents that use it.
+* **Automated review systems**: Build systems that automatically review and suggest content changes, like grammar checks or style improvements.
+* **AI-powered editing**: Make automated suggestions while users are actively editing, helping improve content quality.
+* **Automated publishing**: Prepare and process content for publication, including formatting, metadata updates, and resolving comments.
+
+## Getting started with server-side editor API
+
+This guide explains how to write scripts that can be executed through the Server-side Editor API endpoint. The following sections provide examples of such scripts, each demonstrating a specific use case that can be automated on the server side.
+
+For information about setting up and using the endpoint itself, see the {TODO: link Cloud Services Server-side Editor API} documentation.
```

Comment:
Ideally, we should have a short example here, so all the necessary knowledge is in one place. At least from the API usage perspective – so assuming that someone uploaded a bundle and have credentials already. Something that will be sufficient for a person familiar with the Collab REST APIs already. 

Alternatively/Optionally: This could be a screencast showing one of the usecases: E.g. adding a comment via SS Ed API, while it simultaneously shows up in the editor content in a browser.

Reply:
Note: This screencast will be useful for the blog post as well (cc @wojtekidd)

Reply:
We'll get to it, I believe :)